cmd/webserver.go

```diff
@@ -0,0 +1,202 @@
+				return
+			}
+
+			report, err := webservertype.TypeLaunch(cmd.Context(), config)
+			if err != nil {
+				errorMessage := err.Error()
+				a.OutputSignal.ErrorMessage = &errorMessage
+				a.OutputSignal.Status = 1
+			}
+			a.OutputSignal.Content = report
+
+		},
+	}
+
+	enumerationCmd.Flags().StringSlice("targets", []string{}, "Address of target")
+	enumerationCmd.Flags().String("server", "", "Server type to target (nginx, apache)")
+	enumerationCmd.Flags().Int("timeout", 5000, "Timeout limit in milliseconds")
+	enumerationCmd.Flags().Bool("successfulonly", false, "Only show successful attempts")
+
+	_ = enumerationCmd.MarkFlagRequired("targets")
+	_ = enumerationCmd.MarkFlagRequired("server")
+
+	webServerCmd.AddCommand(enumerationCmd)
+
+	a.RootCmd.AddCommand(webServerCmd)
+
+	validationCmd := &cobra.Command{
+		Use:   "validation",
+		Short: "Preform validation against a specific type of web server",
+		Long:  `Preform validation against a specific type of web server`,
+		Run: func(cmd *cobra.Command, args []string) {
+			// Targets
+			targets, err := cmd.Flags().GetStringSlice("targets")
+			if err != nil {
+				a.OutputSignal.AddError(err)
+				return
+			}
+
+			// ServerType
+			server, err := cmd.Flags().GetString("server")
+			if err != nil {
+				a.OutputSignal.AddError(err)
+				return
+			}
+			serverEnum, err := webscan.NewServerTypeFromString(strings.ToUpper(server))
+			if err != nil {
+				a.OutputSignal.AddError(fmt.Errorf("invalid server type '%s': must be either 'APACHE' or 'NGINX'", server))
+				return
+			}
+
+			// Run Configs
+			timeout, err := cmd.Flags().GetInt("timeout")
+			if err != nil {
+				a.OutputSignal.AddError(err)
+				return
+			}
+			successfulOnly, err := cmd.Flags().GetBool("successfulonly")
+			if err != nil {
+				a.OutputSignal.AddError(err)
+				return
+			}
+			config, err := LoadWebserverTypeConfig(targets, serverEnum, webscan.ProbeTypeValidation, timeout, successfulOnly)
+			if err != nil {
+				a.OutputSignal.AddError(err)
+				return
+			}
+
+			report, err := webservertype.TypeLaunch(cmd.Context(), config)
+			if err != nil {
+				errorMessage := err.Error()
+				a.OutputSignal.ErrorMessage = &errorMessage
+				a.OutputSignal.Status = 1
+			}
+			a.OutputSignal.Content = report
+
+		},
+	}
+
+	validationCmd.Flags().StringSlice("targets", []string{}, "Address of target")
+	validationCmd.Flags().String("server", "", "Server type to target (nginx, apache)")
+	validationCmd.Flags().Int("timeout", 5000, "Timeout limit in milliseconds")
+	validationCmd.Flags().Bool("successfulonly", false, "Only show successful attempts")
+
+	_ = validationCmd.MarkFlagRequired("targets")
+	_ = validationCmd.MarkFlagRequired("server")
+
+	webServerCmd.AddCommand(validationCmd)
+
+	a.RootCmd.AddCommand(webServerCmd)
+}
+
+func LoadWebserverTypeConfig(targets []string, serverEnum webscan.ServerType, probeEnum webscan.ProbeType, timeout int, successfulOnly bool) (*webscan.WebServerTypeConfig, error) {
```

Comment:
nit. From what I've seen in Go, building structs is typically a function that starts with `New`. I also don't actually think we need to worry about the error here. If it's < 1, we should just set it to 0.